File: solutions/0066-plus-one/solution.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        if(digits.size()==0){
            return vector<int>({1});
        }
        int carry = 0;
        if(digits[digits.size()-1]+1 >=10){
            carry = digits[digits.size()-1]-10;
        }
        if(!carry){
            digits[digits.size()-1] +=1;
            return digits;
        }
        vector<int> newv(digits.begin(),digits.end()-1);
        newv = plusOne(newv);
        newv.push_back(0);
        return newv;
    }
};

File: solutions/0066-plus-one/solution.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1
        idx = len(digits)-1
        while idx>=0 and digits[idx]+carry > 9:
            digits[idx] = 0
            carry = 1
            idx = idx-1
        if idx==-1:
            digits.insert(0,1)
        else:
            digits[idx] = digits[idx]+carry
        return digits
        
